### PR TITLE
Feature/rss ecomm 3 14

### PR DESCRIPTION
**Pull request by MariamKochadze**
Deploy: [link](https://deploy-preview-50--garden-with-flowers-test.netlify.app/)

Task: [link](https://github.com/rolling-scopes-school/js-fe-course-en/blob/main/tasks/eCommerce-Application/Sprints/Sprint3/RSS-ECOMM-3_16.md)

Screenshot:
image
score: 10/10
 Implementation Details
User Information: Display the user's first name, last name, and date of birth 📅 on the User Profile page.
User Addresses: List all of the user's saved addresses 🏠 on the User Profile page. Each address entry should display all relevant information (street, city, state, zip code, country). Also, clearly indicate which address is set as the default billing address and which one is the default shipping address. Use appropriate labels or markers to distinguish them.
🎨 Visual Implementation Ideas
User Info Layout: You might want to divide the User Profile page into different sections for clarity. For example, one section could be dedicated to personal information (name, date of birth), while another could list user addresses.
Highlighting Default Addresses: Clearly differentiate the default shipping and billing addresses from the rest. This could be achieved by using different background colors, adding a label, or placing these addresses at the top of the list.
✅ Acceptance Criteria
The User Profile page displays the user's first name, last name, and date of birth.
The User Profile page lists all saved addresses for the user, with all relevant information for each address.
The default billing and shipping addresses are clearly indicated on the User Profile page.

### CODE REVIEW

File: src/api/API.ts [hunk 1]
```diff
@@ -91,6 +92,31 @@ export class API {
...
+  async updateCustomer(customerId: string, updatedCustomerData: CustomerUpdate): Promise<Customer> {
+    let errorMsg = '';
+    const result: Customer = {} as Customer;
+    try {
+      const { body } = await this.client
+        .customers()
+        .withId({ ID: customerId })
+        .post({
+          body: {
+            version: updatedCustomerData.version,
```
Comment (NataliaIv90) [inline src/api/API.ts:106]: good job, Mariam!

File: src/components/profile/Profile.css [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  color: rgb(148, 113, 183);
+}
+
+.profile-info {
+  width: 70%;
+  padding: 1.5rem 1.2rem;
+  border-left: 1px solid grey;
+}
+
+@media screen and (max-width: 480px) {
```
Comment (NataliaIv90) [inline src/components/profile/Profile.css]: It is better to set width 100% for those elements, but place buttons on top and forms on bottom

Comment (NataliaIv90) [reply src/components/profile/Profile.css]: image

File: src/components/profile/Profile.tsx [hunk 3]
```diff
@@ -1,26 +1,43 @@
-import { useAppSelector } from '../../hooks/reduxHooks';
+import { useState } from 'react';
 import { MainWrapper } from '../mainWrapper/MainWrapper';
+import './Profile.css';
+import { ProfileForm } from './profileForm/ProfileForm';
+
+const PROFILE_NAV_ITEMS = [
```
Comment (NataliaIv90) [inline src/components/profile/Profile.tsx:6]: Like this approach!

**Review by NataliaIv90 — COMMENTED** [review]
You've done a great job! I have some questions about responsiveness - I left a comment. Also I don't understand, what does X button do. And for some reason country by default is Georgia. Even if there should be other one.

image

**Review by NataliaIv90 — APPROVED** [review]
Good job, Mariam!

File: src/components/profile/Profile.css (hunk 2 above)
Comment (AbdugafurovaUmida) [reply src/components/profile/Profile.css]: wow,great job Mariam!

**Review by AbdugafurovaUmida — APPROVED** [review]
approved

**Pull request merged**